mu-plugins/blocks/query-filter/src/view.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * WordPress dependencies
+ */
+import { store as wpStore } from '@wordpress/interactivity';
+
+// See https://github.com/WordPress/gutenberg/blob/37f52ae884a40f7cb77ac2484648b4e4ad973b59/packages/block-library/src/navigation/view-interactivity.js
+const focusableSelectors = [
+	'a[href]',
+	'input:not([disabled]):not([type="hidden"]):not([aria-hidden])',
+	'select:not([disabled]):not([aria-hidden])',
+	'textarea:not([disabled]):not([aria-hidden])',
+	'button:not([disabled]):not([aria-hidden])',
+	'[contenteditable]',
+	'[tabindex]:not([tabindex^="-"])',
+];
+
+function closeDropdown( store ) {
+	const { context } = store;
+	context.wporg.queryFilter.isOpen = false;
+	context.wporg.queryFilter.form?.reset();
+
+	const count = context.wporg.queryFilter.form?.querySelectorAll( 'input[type="checkbox"]:checked' ).length;
+	updateToggleLabel( store, count );
+}
+
+function updateToggleLabel( store, count ) {
+	const { context } = store;
+	const toggle = context.wporg.queryFilter.toggleButton;
+	if ( ! toggle ) {
+		return;
+	}
+	toggle.querySelector( 'span' ).innerText = count;
+	if ( count ) {
+		toggle.classList.add( 'has-filter-applied' );
+		toggle.classList.remove( 'has-no-filter-applied' );
+	} else {
+		toggle.classList.add( 'has-no-filter-applied' );
+		toggle.classList.remove( 'has-filter-applied' );
+	}
+}
+
+wpStore( {
+	actions: {
+		wporg: {
+			queryFilter: {
+				toggle: ( store ) => {
+					const { context } = store;
+					if ( context.wporg.queryFilter.isOpen ) {
+						closeDropdown( store );
+					} else {
+						context.wporg.queryFilter.isOpen = true;
+					}
+				},
+				handleFocusout: ( store ) => {
+					const { context, event, ref } = store;
+					if (
+						! context.wporg.queryFilter.hasHover &&
+						! ref.contains( event.relatedTarget ) &&
+						event.target !== window.document.activeElement
+					) {
+						closeDropdown( store );
+					}
+				},
+				handleKeydown: ( store ) => {
+					const { context, event } = store;
+					// If Escape close the dropdown.
+					if ( event.key === 'Escape' ) {
+						closeDropdown( store );
+						context.wporg.queryFilter.toggleButton.focus();
+						return;
+					}
+
+					// Trap focus.
+					if ( event.key === 'Tab' ) {
+						// If shift + tab it change the direction.
+						if (
+							event.shiftKey &&
+							window.document.activeElement === context.wporg.queryFilter.firstFocusableElement
+						) {
+							event.preventDefault();
+							context.wporg.queryFilter.lastFocusableElement.focus();
+						} else if (
+							! event.shiftKey &&
+							window.document.activeElement === context.wporg.queryFilter.lastFocusableElement
+						) {
+							event.preventDefault();
+							context.wporg.queryFilter.firstFocusableElement.focus();
+						}
+					}
+				},
+				handleMouseEnter: ( { context } ) => {
+					context.wporg.queryFilter.hasHover = true;
+				},
+				handleMouseLeave: ( { context } ) => {
+					context.wporg.queryFilter.hasHover = null;
```

Comment:
Is there a reason not to always keep this as a boolean?

Reply:
There wasn't, but I've removed this mouse behavior to fix some other glitches I had noticed https://github.com/WordPress/wporg-mu-plugins/pull/441/commits/4f610db003632cb489041ae76e07a4c585c238fa.